tensorpipe/test/channel/channel_test.cc

```diff
@@ -327,3 +433,106 @@ TEST_P(ChannelTest, contextIsNotJoined) {
         context->createChannel(std::move(conn), Channel::Endpoint::kConnect);
       });
 }
+
+TEST_P(ChannelTest, CallbacksAreDeferred) {
+  // This test wants to make sure that the "heavy lifting" of copying data isn't
+  // performed inline inside the recv method as that would make the user-facing
+  // read method of the pipe blocking.
+  // However, since we can't really check that behavior, we'll check a highly
+  // correlated one: that the recv callback isn't called inline from within the
+  // recv method. We do so by having that behavior cause a deadlock.
+  constexpr auto dataSize = 256;
+
+  // This test is disabled for the CudaIpc channel, as the callback is invoked
+  // inline although the actual heavy lifting happens asynchronously on the
+  // device.
+  if (GetParam()->channelName() == "cuda_ipc") {
```

Comment:
Yeah, this test isn't great... This approach is fine for now, can I just ask you to add a `FIXME` comment on the declaration of the virtual `channelName` (in `channel_test.h`) to say that it's a temporary workaround and that we should try to remove it?